doc/examples/segmentation/plot_rolling_ball.py

```diff
@@ -0,0 +1,148 @@
+"""
+======================================================
+Background Subtraction using the Rolling Ball Method
+======================================================
+
+The rolling ball filter is a segmentation method that aims to separate the
+background from a grayscale image in case of uneven exposure. It is frequently
+used in biomedical image processing and was first proposed by Stanley R.
+Sternberg (1983) in the paper Biomedical Image Processing [1]_.
+
+The idea of the algorithm is quite intuitive. We think of the image as a
+surface that has unit-sized blocks stacked on top of each other for each
+pixel. The number of blocks is determined by the intensity of a pixel. To get
+the intensity of the background at a desired position, we imagine submerging a
+ball into the blocks at the desired pixel position. Once it is completely
+covered by the blocks, the height of the ball determines the intensity of the
+background at that position. We can then *roll* this ball around below the
+surface to get the background values for the entire image. As Sternberg
+puts it: "We can visualize a solid sphere that moves freely within the solid
+volume of the gel image umbra but is constrained by the upper surface of the
+umbra."
+
+.. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1 (1983):
+    22-34. :DOI:`10.1109/MC.1983.1654163`
+
+To implement this algorithm in skimage we can use the following:
+"""
+
+import numpy as np
+from skimage import morphology
+from skimage import data
+import matplotlib.pyplot as plt
+
+
+def rolling_ball(image, radius=30, white_background=False):
+    working_img = image.copy()
+    if white_background:
+        working_img = 255 - working_img
+
+    # tensor representation of the image
+    # (stacked blocks as described in the paper)
+    background = (np.arange(255)[np.newaxis,
```

Comment:
The main issue here is that it assumes an 8-bits image (as far as I understand it), which will fail on other image dtypes.

Reply:
You are correct. The result is a binary 3D np.array with dimension `[height, width, color-depth]`. For a given x, y position and image intensity I(y, x) the array is filled with `background[y, x, 0:I(y, x)] = 1` and 0 otherwise.

It can be generalized to other color-depths than 8-bit, but I am not sure if we can generalize it to float without significant rewriting.

Reply:
Is it required to do input type verification in gallery examples? (I personally think it adds visual clutter that distracts from the important lines of code, but what is the consensus here?)

Reply:
I started to advocate here: https://github.com/scikit-image/scikit-image/issues/3538#issuecomment-663373314 that this function must be included in the library (your work on the example is not lost at all, as we try to illustrate our algorithms in any case). So, to integrate in the library, the function must be as robust as possible. Let's see for other @scikit-image/core   points of view.

Reply:
Yes, for library functions I totally agree that type-checking is necessary. The question then would be if the float should be converted to an int (with precision from a kwarg) and raise a warning, or if the function should throw an exception for float.

Another option could be to rewrite this part to support float, but I am not 100% sure how gray erosion/dilation works with floating-point images.

Reply:
I added code that adjusts to the input image's type. It assumes that the dtype is some form of int though, and may currently cause high memory usage for floats.

Reply:
The current version supports arbitrary dtypes. Floats can be supported by deforming the ball into a spheroid and passing `radius=(spacial_radius, intensity_vertex)` into the function. However, the current version still blocks floats until all the other loose ends are worked out.

Reply:
Fixed; as of figrefoxmetzger/scikit-image@e80a4f3 float (and any other `np.number` type) are valid inputs for `image`. 

It still is kind of strange for float images scaled to `[0, 1]`, so I added a note into the documentation advising to use `rolling_ellipsoid` for non `np.uint8`-types.